ivadomed/utils.py

```diff
@@ -236,14 +239,34 @@ def structurewise_uncertainty(fname_lst, fname_hard, fname_unc_vox, fname_out):
         nib_im = nib.load(fname)
         data_im = nib_im.get_fdata()
         data_lst.append(data_im)
-        data_im_l, _ = label(data_im, structure=bin_struct)
+        data_im_l = []
+        for i in range(data_im.shape[-1]):
+            data_im_tmp, _ = label(data_im[..., i], structure=bin_struct)
+            data_im_l.append(data_im_tmp)
         data_l_lst.append(data_im_l)
         del nib_im
+    data_l_lst = np.array(data_l_lst)
+    # channel went first due to the 'append' function
+    # Befor transpose we are supposed to have MC_simulation X Channel X Height X Width X Depth
+    # After transpose we expect to have MC_simulation X Height X Width X Depth X Channel
+    # this is wanted because the network output has a shape of Height X Width X Depth X Channel
+    # So the rest of the code is designed to use something like this
+    if len(data_l_lst.shape) == 4:
+        data_l_lst = np.transpose(data_l_lst, (0, 2, 3, 4, 1))
+    elif len(data_l_lst.shape) == 3:
+        data_l_lst = np.transpose(data_l_lst, (0, 2, 3, 1))
```

Comment:
I would add a "else" as safety.

Reply:
Done in ed6cf059e810dbd62b1742659c084cc131105587